sklearn/decomposition/_dict_learning.py

```diff
@@ -1227,7 +1227,7 @@ def dict_learning(
         positive_code=positive_code,
         positive_dict=positive_dict,
         transform_max_iter=method_max_iter,
-    )
+    ).set_output(transform="default")
```

Comment:
This avoids calling svd_flip on a pandas dataframe in the subsequent call to `fit_transform`.

Reply:
Now I am thinking that since all our function are wrapping an estimator, we should most probably use this pattern everywhere. Anyway not link to the current PR.